src/Types.jl

```diff
@@ -337,6 +337,8 @@ function EnvCache(env::Union{Nothing,String}=nothing)
     write_env_usage(manifest_file, "manifest_usage.toml")
     manifest = read_manifest(manifest_file)
 
+    Types.check_warn_manifest_julia_version_compat(manifest, manifest_file)
```

Comment:
I don't think this one should be here. This will trigger even when doing "API" style use (like in Pluto).

Reply:
I removed this.

Reply:
Oh, gotta update test as well :)

Reply:
Ok so this PR now only warns during instantiate. It's obviously better than before, but not sure how many manifest compat issues it will catch given instantiate isn't that common, I believe